tests/test_selenium.py

```diff
@@ -53,6 +75,350 @@ def test_signup(selenium, live_server, client):
-    live_server.thread.terminate()
-    live_server.thread.join()
+
+
+@mock_plaid_balance
+@mock_plaid_accounts
+@mock_plaid_transactions
+@pytest.mark.django_db(transaction=True)
+def test_add_bucket(selenium, live_server, client):
+    """
+    Tests adding a bucket
+    """
+    user = User.objects.create_user('temporary', 'temporary@gmail.com', 'temporary')
+    user.save()
+    user.userbank.create(
+        item_id='dummy1', access_token='dummy2',
+        institution_name='dummy3', current_balance_field=0,
+        account_name_field="dummy4", income_field=0,
+        expenditure_field=0
+    )
+    client.login(username='temporary', password='temporary')
+    assert user.profile.owned_bucket.count() == 0
+    cookie = client.cookies['sessionid']
+    selenium.get('%s%s' % (live_server, '/login'))
+    selenium.add_cookie({
+        'name': 'sessionid',
+        'value': cookie.value,
+        'secure': False,
+        'path': '/',
+    })
+    selenium.get('%s%s' % (live_server, '/home'))
+    selenium.implicitly_wait(30)
+    newbuck = selenium.find_element_by_xpath("//button[contains(.,'New')]")
+    newbuck.click()
+    cancel = selenium.find_element_by_id("cancel")
+    cancel.click()
+    newbuck = selenium.find_element_by_xpath("//button[contains(.,'New')]")
+    newbuck.click()
+    buckname = selenium.find_element_by_id("name")
+    buckname.send_keys("IAMATESTBUCKET")
+    invest = selenium.find_element_by_id("investment")
+    invest.send_keys("5000")
+    save = selenium.find_element_by_id("save")
+    save.click()
+    assert user.profile.owned_bucket.count() == 1
+
+
+@mock_plaid_balance
+@mock_plaid_accounts
+@mock_plaid_transactions
+@pytest.mark.django_db(transaction=True)
+def test_delete_bucket(selenium, live_server, client):
+    """
+    Tests deleting a bucket
+    """
+    user = User.objects.create_user('temporary', 'temporary@gmail.com', 'temporary')
+    user.save()
+    user.userbank.create(
+        item_id='dummy1', access_token='dummy2',
+        institution_name='dummy3', current_balance_field=0,
+        account_name_field="dummy4", income_field=0,
+        expenditure_field=0
+    )
+    client.login(username='temporary', password='temporary')
+    assert user.profile.owned_bucket.count() == 0
+    cookie = client.cookies['sessionid']
+    buck = InvestmentBucket.create_new_bucket(
+        name="IAMATESTBUCKET", public=True,
+        owner=user.profile)
+    buck.save()
+    assert user.profile.owned_bucket.count() == 1
+    selenium.get('%s%s' % (live_server, '/login'))
+    selenium.add_cookie({
+        'name': 'sessionid',
+        'value': cookie.value,
+        'secure': False,
+        'path': '/',
+    })
+    selenium.get('%s%s' % (live_server, '/home'))
+    selenium.implicitly_wait(30)
+    delete_button = selenium.find_element_by_id("delete")
+    delete_button.click()
+    cancel_delete = selenium.find_element_by_id("keep")
+    cancel_delete.click()
+    time.sleep(1)
```

Comment:
Why?

Reply:
because although the delete works on the front end, if you immediately try to assert the delete, it may not have propagated to the db yet in which case the test fails.

Reply:
Not a big fan of that solution, since waiting is generally flaky. Can you create two buckets, cancel on one and delete the other one? Then you can assert that the correct one is still there, while the wrong one is deleted

Reply:
Actually wait, this was because without the sleep the modal isn't dismissed by the time the button is clicked, so the button is not clicked properly

Reply:
Then check whether the modal exists like done here https://github.com/Neitsch/ASE4156/blob/master/tests/test_selenium.py#L52

Reply:
The problem is the button still exists, its just under the modal. So I can't wait for it to appear because its already there, its just unclickable. Is there some way to test if its clickable? I can't find one.

Reply:
Nobody will notice I guess